server/commands/check.go

```diff
@@ -214,16 +214,17 @@ func (q *CheckQuery) resolveDirectUserSet(
 	rc *resolutionContext,
 	typesys *typesystem.TypeSystem,
 ) error {
-	done := make(chan struct{})
-	defer close(done)
-
 	var wg sync.WaitGroup
-	c := make(chan *chanResolveResult)
+	c := make(chan *chanResolveResult, 2)
```

Comment:
Since now we launch two goroutines, leave enough space in the channel for both to write in it. If we leave this as is, it is possible for the 1st goroutine to write to it and the 2nd goroutine to also attempt a write and becoming blocked.